contracts/v3/FiatTokenV3.sol

```diff
@@ -0,0 +1,300 @@
+        emit Blacklisted(_account);
+    }
+
+    /**
+     * @dev Removes account from blacklist
+     * @param _account The address to remove from the blacklist
+     */
+    function unBlacklist(address _account) override external onlyBlacklister {
+       _unBlacklist(_account);
+    }
+
+    /**
+     * @dev Internal function to process removals from the blacklist
+     * @param _account The address to remove from the blacklist
+     */
+    function _unBlacklist(address _account) internal {
+        balances[_account] = balances[_account] & ~(uint256(1) << 255);
+        emit UnBlacklisted(_account);
+    }
+
+    /**
+     * @dev Checks if account is blacklisted
+     * @param _account The address to check
+     */
+    function isBlacklisted(address _account) override external view returns (bool) {
+        return balances[_account] >> 255 == 1;
+    }
+
+    /**
+     * @dev Throws if argument account is blacklisted
+     * @param _account The address to check
+     */
+    modifier notBlacklisted(address _account) override {
+        require(
+            !(balances[_account] >> 255 == 1),
+            "Blacklistable: account is blacklisted"
+        );
+        _;
+    }
+
+    /**
+     * @dev Get token balance of an account
+     * @param account address The account
+     */
+    function balanceOf(address account)
+        external
+        override
+        view
+        returns (uint256)
+    {
+        // return balance without effect of blacklist high bit
+        return balances[account] & ~(uint256(1) << 255);
+    }
+
+    /**
+     * @dev Get token balance of an account
+     * @param account address The account
+     */
+    function _getBalanceAndBlacklistStatus(address account)
+        internal
+        view
+        returns (uint256, bool)
+    {
+        uint256 balance = balances[account];
+        return (balance & ~(uint256(1) << 255), balance >> 255 == 1);
+    }
+
+    /**
+     * @notice Internal function to process transfers
+     * @param from  Payer's address
+     * @param to    Payee's address
+     * @param value Transfer amount
+     */
+    function _transfer(
+        address from,
+        address to,
+        uint256 value
+    ) internal override {
+        require(from != address(0), "ERC20: transfer from the zero address");
+        require(to != address(0), "ERC20: transfer to the zero address");
+
+        (uint256 fromBalance, bool isFromBlacklisted) = _getBalanceAndBlacklistStatus(from);
+
+        require(!isFromBlacklisted, "Blacklistable: account is blacklisted");
+
+        require(
+            value <= fromBalance,
+            "ERC20: transfer amount exceeds balance"
+        );
+
+        balances[from] = fromBalance.sub(value);
+
+        (uint256 toBalance, bool isToBlacklisted) = _getBalanceAndBlacklistStatus(to);
+        require(!isToBlacklisted, "Blacklistable: account is blacklisted");
+
+        balances[to] = toBalance.add(value);
```

Comment:
Yes it would, the balance required to do so is outlandishly large though (5.78 * 10^69 USD).

The current total supply of USDC (in 256-bit form) is:

`0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010100010001010001001111111100010101011110111111100101110`

Reply:
This risk can be completely eliminated if the mint function caps the total supply of the token so that the high bit is guaranteed to be unused. This adds no additional cost to transfers.

Reply:
that's a good idea!

Reply:
implemented @frangio's idea

https://github.com/centrehq/centre-tokens/pull/357/commits/4400ddaba67401b669d5396b8a5a965761e9338d